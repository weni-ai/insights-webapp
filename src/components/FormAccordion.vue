<template>
  <UnnnicCollapse
    class="form-accordion"
    :class="{
      'form-accordion--active': active,
      highlighted: active && highlighted,
    }"
    :active="active"
    @change="$emit('update:active', $event)"
  >
    <template #header>
      <header class="form-accordion__header">
        <UnnnicIcon
          icon="check_circle"
          :scheme="iconScheme"
          size="avatar-nano"
          filled
        />
        {{ title }}
      </header>
    </template>
    <slot name="content"></slot>
  </UnnnicCollapse>
</template>

<script>
export default {
  name: 'FormAccordion',

  props: {
    active: {
      type: Boolean,
      default: false,
    },
    title: {
      type: String,
      default: '',
    },
    validConfig: {
      type: Boolean,
      default: false,
    },
    highlighted: {
      type: Boolean,
      default: false,
    },
  },

  emits: ['update:active'],

  computed: {
    iconScheme() {
      return this.validConfig ? 'weni-600' : 'neutral-soft';
    },
  },
};
</script>

<style lang="scss" scoped>
.form-accordion {
  border-radius: $unnnic-border-radius-sm;
  border: $unnnic-border-width-thinner solid $unnnic-color-neutral-soft;

  &--active.highlighted {
    background-color: $unnnic-color-weni-50;
    border: 1px solid $unnnic-color-weni-500;
  }

  :deep(.unnnic-collapse__header) {
    padding: $unnnic-spacing-ant;
  }

  :deep(.unnnic-collapse__body) {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
  }

  &__header {
    display: flex;
    align-items: center;
    gap: $unnnic-spacing-nano;

    color: $unnnic-color-neutral-dark;
    font-size: $unnnic-font-size-body-gt;
    font-weight: $unnnic-font-weight-bold;
    line-height: $unnnic-line-height-small * 5.5;
  }
}
</style>
