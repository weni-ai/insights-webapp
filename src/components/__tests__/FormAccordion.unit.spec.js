import { beforeEach, describe, expect, it } from 'vitest';
import { mount } from '@vue/test-utils';
import FormAccordion from '@/components/FormAccordion.vue';

describe('FormAccordion.vue', () => {
  let wrapper;

  const createWrapper = (props = {}) => {
    return mount(FormAccordion, {
      props,
      global: {
        stubs: {
          UnnnicCollapse: true,
          UnnnicIcon: true,
        },
      },
      slots: {
        content: '<div class="slot-content">Slot Content</div>',
      },
    });
  };

  beforeEach(() => {
    wrapper = createWrapper();
  });

  describe('Props', () => {
    it('should render title text', () => {
      wrapper = createWrapper({ title: 'Test Title' });
      const header = wrapper.find('.form-accordion__header');

      expect(header.text()).toContain('Test Title');
    });

    it('should apply active and highlighted classes when active and highlighted are true', () => {
      wrapper = createWrapper({ active: true, highlighted: true });
      expect(wrapper.find('.form-accordion').classes()).toContain(
        'form-accordion--active',
      );
      expect(wrapper.find('.form-accordion').classes()).toContain(
        'highlighted',
      );
    });

    it('should not apply active or highlighted classes when props are false', () => {
      wrapper = createWrapper({ active: false, highlighted: false });
      expect(wrapper.find('.form-accordion').classes()).not.toContain(
        'form-accordion--active',
      );
      expect(wrapper.find('.form-accordion').classes()).not.toContain(
        'highlighted',
      );
    });
  });

  describe('Computed Properties', () => {
    it('iconScheme should return "weni-600" if validConfig is true', () => {
      wrapper = createWrapper({ validConfig: true });
      expect(wrapper.vm.iconScheme).toBe('weni-600');
    });

    it('iconScheme should return "neutral-soft" if validConfig is false', () => {
      wrapper = createWrapper({ validConfig: false });
      expect(wrapper.vm.iconScheme).toBe('neutral-soft');
    });
  });

  describe('Emits', () => {
    it('should emit update:active event with new value when change event is triggered on UnnnicCollapse', async () => {
      const collapseComponent = wrapper.findComponent(
        '[data-test-id="form-accordion"]',
      );
      await collapseComponent.vm.$emit('change', true);

      expect(wrapper.emitted('update:active')).toBeTruthy();
      expect(wrapper.emitted('update:active')[0]).toEqual([true]);
    });
  });

  describe('Slots', () => {
    it('should render slot content in the content slot', () => {
      expect(wrapper.find('.slot-content').exists()).toBe(true);
      expect(wrapper.find('.slot-content').text()).toBe('Slot Content');
    });
  });

  describe('Icon', () => {
    it('should render UnnnicIcon with correct icon, scheme, size, and filled attributes', () => {
      wrapper = createWrapper({ validConfig: true });
      const icon = wrapper.findComponent('[data-test-id="check_circle"]');

      expect(icon.exists()).toBe(true);
      expect(icon.props('icon')).toBe('check_circle');
      expect(icon.props('scheme')).toBe('weni-600');
      expect(icon.props('size')).toBe('avatar-nano');
      expect(icon.props('filled')).toBe(true);
    });
  });

  it('should match the snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
