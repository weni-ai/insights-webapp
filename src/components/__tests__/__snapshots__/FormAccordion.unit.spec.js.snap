// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FormAccordion.vue > should match the snapshot 1`] = `
<div
  class="unnnic-collapse unnnic-collapse--size-lg form-accordion"
  data-test-id="form-accordion"
  data-testid="collapse"
  data-v-085a4372=""
  data-v-1f77973f=""
>
  <div
    class="unnnic-collapse__header"
    data-testid="collapse-header"
    data-v-1f77973f=""
  >
    
    
    <header
      class="form-accordion__header"
      data-v-085a4372=""
    >
      <unnnic-icon-stub
        clickable="false"
        data-test-id="check_circle"
        data-v-085a4372=""
        filled="true"
        icon="check_circle"
        next="false"
        scheme="neutral-soft"
        size="avatar-nano"
      />
       
    </header>
    
    <unnnic-icon-stub
      clickable="false"
      data-v-1f77973f=""
      filled="false"
      icon="arrow-button-down-1"
      next="false"
      scheme="neutral-cloudy"
      size="xs"
    />
  </div>
  <div
    class="unnnic-collapse__body"
    data-v-1f77973f=""
    style="display: none;"
  >
    
    
    <div
      class="slot-content"
    >
      Slot Content
    </div>
    
    
  </div>
</div>
`;
